### ISSUE TITLE
XXL Arrays

### ISSUE THREAD

**Issue by Dylan-DPC** [rust-lang/rust#71699]
### Code

```Rust
fn foo([(), .., ()]: [(); 1 << 40]) {}

fn main() {
    foo([(); 1 << 40]);
}
```
### Error output

```
thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: TryFromIntError(())', src/librustc_mir_build/build/matches/util.rs:36:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

**Comment by spastorino** [comment]
Assigning `P-medium` as [discussed as part of the Prioritization Working Group process](https://rust-lang.zulipchat.com/#narrow/stream/227806-t-compiler.2Fwg-prioritization/topic/I-prioritize.20.2371699.20XXL.20Arrays) and removing `I-prioritize`.

**Comment by DutchGhost** [comment]
This can be shortened to:
```Rust
fn bug([..]: [(); !0]) {}
```

**Comment by Alexendoo** [comment]
Fixed by #75893
